tests/test_triso/test_triso.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import glob
+import random
+from math import sqrt
+
+import numpy as np
+
+sys.path.insert(0, os.pardir)
+from testing_harness import PyAPITestHarness
+import openmc
+import openmc.model
+
+
+class TRISOTestHarness(PyAPITestHarness):
+    def _build_inputs(self):
+        # Define TRISO matrials
+        fuel = openmc.Material()
+        fuel.set_density('g/cm3', 10.5)
+        fuel.add_nuclide('U-235', 0.14154)
+        fuel.add_nuclide('U-238', 0.85846)
+        fuel.add_nuclide('C-Nat', 0.5)
+        fuel.add_nuclide('O-16', 1.5)
+
+        porous_carbon = openmc.Material()
+        porous_carbon.set_density('g/cm3', 1.0)
+        porous_carbon.add_nuclide('C-Nat', 1.0)
+        porous_carbon.add_s_alpha_beta('Graph', '71t')
+
+        ipyc = openmc.Material()
+        ipyc.set_density('g/cm3', 1.90)
+        ipyc.add_nuclide('C-Nat', 1.0)
+        ipyc.add_s_alpha_beta('Graph', '71t')
+
+        sic = openmc.Material()
+        sic.set_density('g/cm3', 3.20)
+        sic.add_element('Si', 1.0)
+        sic.add_nuclide('C-Nat', 1.0)
+
+        opyc = openmc.Material()
+        opyc.set_density('g/cm3', 1.87)
+        opyc.add_nuclide('C-Nat', 1.0)
+        opyc.add_s_alpha_beta('Graph', '71t')
+
+        graphite = openmc.Material()
+        graphite.set_density('g/cm3', 1.1995)
+        graphite.add_nuclide('C-Nat', 1.0)
+        graphite.add_s_alpha_beta('Graph', '71t')
+
+        # Create TRISO particles
+        materials = [fuel, porous_carbon, ipyc, sic, opyc]
+        radii = np.array([212.5, 312.5, 347.5, 382.5, 422.5])*1e-4
+        trisos = []
+        random.seed(1)
+        for i in range(100):
+            # Randomly sample location
+            x = random.uniform(-0.5, 0.5)
+            y = random.uniform(-0.5, 0.5)
+            z = random.uniform(-0.5, 0.5)
+            t = openmc.model.TRISO(materials, radii, (x, y, z))
+
+            # Make sure TRISO doesn't overlap with another
+            for tp in trisos:
+                xp, yp, zp = tp.center
+                distance = sqrt((x - xp)**2 + (y - yp)**2 + (z - zp)**2)
+                if distance <= 2*radii[-1]:
+                    break
+            else:
+                trisos.append(t)
+
+        # Define box to contain lattice
+        min_x = openmc.XPlane(x0=-0.5, boundary_type='reflective')
+        max_x = openmc.XPlane(x0=0.5, boundary_type='reflective')
+        min_y = openmc.YPlane(y0=-0.5, boundary_type='reflective')
+        max_y = openmc.YPlane(y0=0.5, boundary_type='reflective')
+        min_z = openmc.ZPlane(z0=-0.5, boundary_type='reflective')
+        max_z = openmc.ZPlane(z0=0.5, boundary_type='reflective')
+        box = openmc.Cell(region=+min_x & -max_x & +min_y & -max_y & +min_z & -max_z)
+
+        # Create lattice
+        ll, ur = box.region.bounding_box
+        shape = (3, 3, 3)
+        lattice = openmc.model.create_triso_lattice(
+            trisos, ll, (ur - ll)/shape, shape, graphite)
```

Comment:
Bearing in mind that our test suite is a big part of our documentation, I would recommend making it more clear that this is a lattice pitch
```python
        pitch = (ur - ll) / shape
        lattice = openmc.model.create_triso_lattice(
            trisos, ll, pitch, shape, graphite)
```